pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -297,7 +335,13 @@ func (c *Cluster) makeDeployment(nodeName string, selection rookalpha.Selection,
 					},
 				},
 				Spec: v1.PodSpec{
-					NodeSelector:       map[string]string{v1.LabelHostname: nodeName},
+					Affinity: &v1.Affinity{
+						NodeAffinity:    osdProps.placement.NodeAffinity,
```

Comment:
If the affinity is not set in the manifest, will this have an issue setting null affinity? I believe it will be fine, but just checking.

Reply:
cc @ashishranjan738

Reply:
we don't need this as there is a placement function getting called after this